### DISCUSSION TITLE
Moving notes around / deleting notes

### DISCUSSION THREAD

**Discussion by caenrique** [zk-org/zk-nvim#177] [Help]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

I'm looking for a simple way to move/delete notes using some of the built in features like the note picker. Ideally this is the workflows I want:

- I search for a note using `:ZkNotes`
- after filtering in the picker (in my case telescope) I select one or many entries
- I have a keymapping to delete those notes or move them to a new directory (being prompted for the destination)

another one would be as simple as deleting the current open note.

One thing that complicates this a bit more is the possible broken links to the deleted/moved note.

How is everyone handling this?

**Comment by tjex** [comment]
Currently there is no in-built functionality for moving / deleting notes and dealing with broken links. But there has been some serious thought into it in the past, [see here](https://github.com/zk-org/zk/issues/200#issuecomment-1126803588). 
It would most certainly be an excellent feature to have. Hopefully we can implement this in the not too distant future.

Personally, I never delete notes unless they're new / being made fresh and I make a mistake or something.
One way to handle it would be to `sed` replace the filepath of a moved note with the new file path, or if the note is deleted to replace the filepath with "DELETED-NOTE", so that when your LSP complains, you know its because of the deleted note and you can safely remove the link symbols.

**Comment by dzintars** [comment]
I know, this is not the answer... but, this is one of the reason why I use flat file structure.
![image](https://github.com/user-attachments/assets/084145b4-3209-4b7d-81f0-d89acaf3b45a)

This way I don't never need to move notes around. If I want to split it, then I just create new note and leave the old one. Rarely I just use `:call delete(expand('%')) | bdelete!` to delete the current note.
